File: back-end/internal/interface/dto/income.go
package dto

// Income はAPIレスポンス用の収入DTOを表す。
type Income struct {
	Memo              *string `json:"memo"`
	ID                string  `json:"id"`
	IncomeDate        string  `json:"incomeDate"`
	Amount            int     `json:"amount"`
	IncludedInBalance bool    `json:"includedInBalance"`
}

// CreateIncomeRequest は収入登録requestを表す。
type CreateIncomeRequest struct {
	IncludedInBalance *bool   `json:"includedInBalance"`
	Memo              *string `json:"memo"`
	IncomeDate        string  `json:"incomeDate"`
	Amount            int     `json:"amount"`
}

// UpdateIncomeRequest は収入更新requestを表す。
type UpdateIncomeRequest struct {
	IncludedInBalance *bool   `json:"includedInBalance"`
	Memo              *string `json:"memo"`
	IncomeDate        string  `json:"incomeDate"`
	Amount            int     `json:"amount"`
}

// IncomeResponse は単一収入responseを表す。
type IncomeResponse struct {
	Income Income `json:"income"`
}

// ListIncomesResponse は収入一覧responseを表す。
type ListIncomesResponse struct {
	Items []Income `json:"items"`
}

File: front-end/src/domains/income.ts
/**
 * 収入domain model。
 */
export type Income = {
  /** 収入金額 */
  readonly amount: number;
  /** 収入ID */
  readonly id: string;
  /** 今月使えるお金に含めるか */
  readonly includedInBalance: boolean;
  /** YYYY-MM-DD形式の入金日 */
  readonly incomeDate: string;
  /** 収入メモ */
  readonly memo: null | string;
};
